src/simd/Mallocator.hpp

```diff
@@ -42,6 +42,8 @@ struct Mallocator
 
   T* allocate(std::size_t n)
   {
+    if (n == 0)
+      return nullptr;
```

Comment:
My understanding for C++ allocator is allocate(0) is unspecified behavior.
All the containers we made must not rely on unspecified behavior.
I feel better throw an error if 0 is requested. This helps catching any bugs in our containers and the performance hit due to the check should be negligible. 
The check needs to be added in both allocate() and deallocate()
@PDoakORNL any thoughts?

Reply:
I read it as unspecified but you aren't allowed to crash. I wasn't ready to go to a throw, but I think you're right. A 0  allocate shouldn't make it to Mallocator, it might not be safe with another allocator and we don't know what they will do. I will update.

Reply:
So Mallocator will throw now.